docs/WIA.md

```diff
@@ -0,0 +1,234 @@
+ - For the compression methods NONE and PURGE, if the end offset of the last ``wia_except_list_t`` is not evenly divisible by 4, padding is inserted after it so that the data afterwards will start at a 4 byte boundary. This padding is not inserted for the other compression methods.
+
+## `wia_segment_t`
+
+This struct is used by the simple compression method PURGE, which stores runs of zeroes efficiently and stores other data as is.
+
+|Type and name|Description|
+|--|--|
+|`u32 offset`|The offset of `data` within the decompressed data. (Any `wia_except_list_t` structs are not counted as part of the decompressed data.)|
+|`u32 size`|The number of bytes in `data`.|
+|`u8 data[size]`|Data.|
+
+Each PURGE chunk contains one or more `wia_segment_t` structs stored in order of ascending `offset`, followed by a SHA-1 hash (0x14 bytes) of the `wia_except_list_t` structs (if any) and the `wia_segment_t` structs. Bytes in the decompressed data that are not covered by any `wia_segment_t` struct are set to `0x00`.
+
+# RVZ file format description
+
+RVZ is a file format which is closely based on WIA. The differences are as follows:
+
+* Zstandard has been added as a compression method. `compression` in `wia_disc_t` is set to 5 when Zstandard is used, and there is no compressor specific data. `compr_level` in `wia_disc_t` should be treated as signed instead of unsigned because Zstandard supports negative compression levels.
+* PURGE has been removed as a compression method.
+* Chunk sizes smaller than 2 MiB are supported. The following applies when using a chunk size smaller than 2 MiB:
+    * The chunk size must be at least 32 KiB and must be a power of two. (Just like with WIA, sizes larger than 2 MiB do not have to be a power of two, they just have to be an integer multiple of 2 MiB.)
+    * For Wii partition data, each chunk contains one `wia_except_list_t` which contains exceptions for that chunk (and no other chunks). Offset 0 refers to the first hash of the current chunk, not the first hash of the full 2 MiB of data.
+* An encoding scheme which is described below is used to store pseudorandom padding data losslessly.
+
+## RVZ packing
+
+The RVZ packing encoding scheme is applied to all `wia_group_t` data, with any bzip2/LZMA/Zstandard compression being applied on top of it. (In other words, when reading an RVZ file, bzip2/LZMA/Zstandard decompression is done before decoding the RVZ packing.) RVZ packed data can be decoded as follows:
+
+1. Read 4 bytes of data and interpret it as a 32-bit unsigned big endian integer. Call this `size`.
+2. If the most significant bit of `size` is not set, read `size` bytes and output them unchanged. If the most significant bit of `size` is set, unset the most significant bit of `size`, then read 68 bytes of PRNG seed data and output `size` bytes using the PRNG algorithm described below.
```

Comment:
How often are there more than one PRNG data region within one 32 kB block? And if so, do they sometimes share the same seed?

Reply:
There are essentially two reasons why you might have more than one PRNG data region within one 32 KiB block:

1. The Wii 32 KiB vs 31 KiB mismatch. The vast majority of Wii partition blocks (62 out of 64) contain more than one PRNG data region because of this. The seeds are different in this case.

2. The case where a block contains a file followed by PRNG data followed by another file followed by more PRNG data. If all this happens within 32 KiB, you will have two PRNG data regions with the same seed. I don't know how common this is.